### PR TITLE
(feat: stateful) return distinct changes and filter undefined state

### PR DESCRIPTION
**Pull request by hoebbelsB**
before this change the stateful pipe would not have emitted distinct values, since the distinctUntilChanged pipe was applied before the mapping. furthermore undefined values would have been emitted which shouldn't be the case

### CODE REVIEW

File: libs/rxjs-state/src/lib/core/operators/stateful.ts [hunk 1]
```diff
@@ -33,13 +33,14 @@ export function stateful<T, A, B, C, D, E>(
 export function stateful<T, R>(...optionalDerive: OperatorFunction<T, R>[]): OperatorFunction<T, T | R> {
```
Comment (BioPhoton) [inline libs/rxjs-state/src/lib/core/operators/stateful.ts]: We should add a very detailed description of why each line is doing.

Some thoughts here:
```typescript
// distinct same state objects (e.g. a default emission of default switch cases, incorrect mutable handling of data)
// @TODO evaluate benefits vs. overhead
distinctUntilChanged(),
// CUSTOM LOGIC HERE
pipe(...customOperations),
// initial emissions, undefined is no state, pollution with skip(1)
filter(v => v!== undefined),
// distinct same derivation value
distinctUntilChanged(),
// reuse custom operations result for multiple subscribers and reemit the last canculated value.
shareReplay({bufferSize: 1, refCount: true})
```

File: libs/rxjs-state/src/lib/core/operators/stateful.ts [hunk 2]
```diff
@@ -33,13 +33,14 @@ export function stateful<T, A, B, C, D, E>(
...
     return (s: Observable<T>): Observable<T | R> => {
         return s.pipe(
-            distinctUntilChanged(),
```
Comment (BioPhoton) [inline libs/rxjs-state/src/lib/core/operators/stateful.ts]: we could keep it for above-explained reasons?

File: libs/rxjs-state/src/lib/core/operators/stateful.ts (hunk 1 above)
Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/stateful.ts]: thx, implemented it right away

File: libs/rxjs-state/src/lib/core/operators/stateful.ts (hunk 2 above)
Comment (hoebbelsB) [reply libs/rxjs-state/src/lib/core/operators/stateful.ts]: let's try it!

**Pull request merged**